### PR TITLE
Beoble 1830 chatroom edit info panel

### PR DESCRIPTION
**Pull request by chanhyeokryptonium**

### CODE REVIEW

File: packages/beoble/components/OrganSystems/ProfileUpdate/index.tsx [hunk 1]
```diff
@@ -0,0 +1,170 @@
+import { INFT } from '@beoble/js-sdk';
```
Comment (chanhyeokryptonium) [inline packages/beoble/components/OrganSystems/ProfileUpdate/index.tsx:1]: new OrganSystem ProfileUpdate - to be used for both user + chatroom

File: packages/beoble/components/OrganSystems/ProfileUpdate/index.tsx [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+                helperText={errors[inputFormDatum.name]?.message as string}
+                error={!!errors[inputFormDatum.name]}
+                defaultValue={inputFormDatum.defaultValue}
+                {...InputStyle}
+              />
+            ))}
+
+            {children}
+          </InputForm>
+          <PublishButtonContainer>
```
Comment (bamnenim-kryptonium) [inline packages/beoble/components/OrganSystems/ProfileUpdate/index.tsx:152]: roll back to submit button inside of the form.
instead, use custom FormEventHandler

File: packages/beoble/components/OrganSystems/ProfileUpdate/index.tsx [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+      </ChatroomImageContainer>
+      {editPage === 'DEFAULT' ? (
+        <PagePaddingContainer>
+          <InputForm>
+            {inputFormData.map((inputFormDatum) => (
+              <InputField
+                key={inputFormDatum.name}
+                {...register(inputFormDatum.name)}
+                title={inputFormDatum.title}
+                required={inputFormDatum.required ?? false}
```
Comment (bamnenim-kryptonium) [inline packages/beoble/components/OrganSystems/ProfileUpdate/index.tsx:141]: just pass required

File: packages/beoble/components/OrganSystems/ProfileUpdate/index.tsx [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+  ) => Promise<void>;
+}
+
+export const ProfileUpdate = ({
+  children,
+  editPage,
+  setEditPage,
+  inputFormSchema,
+  inputFormData,
+  avatarPreviewHandler,
```
Comment (bamnenim-kryptonium) [inline packages/beoble/components/OrganSystems/ProfileUpdate/index.tsx:81]: pass default image

File: packages/react/src/hooks/mutations/chatroom/usePutChatroom.ts [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+              : 'background_profile'
+          ] = {
+            image_url: newImageUrl,
+            nft: newNft,
+          };
+        }
+        replace(optimisticChatroom);
+        return { chatroom };
+      },
+      onError: (error, variables, context: any) => {
```
Comment (bamnenim-kryptonium) [inline packages/react/src/hooks/mutations/chatroom/usePutChatroom.ts:78]: no type auto-inferred?

File: packages/react/src/hooks/mutations/chatroom/usePutChatroomSettings.ts [hunk 6]
```diff
@@ -24,16 +32,34 @@ export default function usePutChatroomSettings() {
...
+        }
+        replace(optimisticChatroom);
+        return { chatroom };
+      },
       onSuccess: (data) => {
         queryClient.invalidateQueries({
           queryKey: [QueryKeys.chatrooms, data.id],
         });
       },
+      onError: (error, variables, context: any) => {
```
Comment (bamnenim-kryptonium) [inline packages/react/src/hooks/mutations/chatroom/usePutChatroomSettings.ts:59]: context type inferred?

File: packages/react/src/hooks/mutations/chatroom/usePutChatroomSettings.ts [hunk 7]
```diff
@@ -24,16 +32,34 @@ export default function usePutChatroomSettings() {
...
+          optimisticChatroom.chatroom_setting.web3_access_setting.access_map.REQUEST_JOIN_CHATROOM =
+            {
+              input_rule: newInputRule,
+            };
+        }
+        replace(optimisticChatroom);
+        return { chatroom };
+      },
       onSuccess: (data) => {
         queryClient.invalidateQueries({
```
Comment (bamnenim-kryptonium) [inline packages/react/src/hooks/mutations/chatroom/usePutChatroomSettings.ts:55]: change queryKey to this [QueryKeys.chatroomsByUserId, userId]

File: packages/react/src/hooks/mutations/user/usePutUserChatroomSetting.ts
Comment (bamnenim-kryptonium) [inline packages/react/src/hooks/mutations/user/usePutUserChatroomSetting.ts:1]: queryClient.invalidateQueries({
          queryKey: [QueryKeys.userChatroomSettings, user_id],
        });

File: packages/js-sdk/src/lib/Responses/user.ts [hunk 8]
```diff
@@ -128,7 +129,15 @@ export interface IUserWeb3InfoChainSnapshot {
   };
   non_fungible_tokens: IWeb3InfoRecord<INFT>[];
   fungible_tokens: IFungibleToken[];
-  transactions: ITransaction[];
+  transactions: IAllTransaction;
```
Comment (bamnenim-kryptonium) [inline packages/js-sdk/src/lib/Responses/user.ts:132]: IWeb3InfoRecord<ITransaction[]>

File: packages/beoble/components/Organisms/PanelPages/types/index.ts [hunk 9]
```diff
@@ -40,11 +41,18 @@ type FollowQuery = {
   userId: string;
 };
 
+type ChatroomFilterEditQUery = {
+  destination: 'DEFAULT' | 'NFT_FILTER' | 'TOKEN_FILTER' | 'TXN_FILTER';
```
Comment (bamnenim-kryptonium) [inline packages/beoble/components/Organisms/PanelPages/types/index.ts:45]: define type separately to reuse

File: packages/beoble/consts/addChatFilterItems.ts [hunk 10]
```diff
@@ -5,16 +5,19 @@ export const addChatFilterItems = {
     addFilterIconImage: 'green-plus.png',
     header: 'NFT rule',
     type: Web3FilterTypes.nftOwnership,
```
Comment (bamnenim-kryptonium) [inline packages/beoble/consts/addChatFilterItems.ts:7]: like here

File: packages/beoble/components/Organisms/PanelPages/RightPanels/ChatroomInfo/index.tsx [hunk 11]
```diff
@@ -174,23 +218,127 @@ export const ChatroomInfoPage = ({ id }: PanelPageProps) => {
...
 
   return (
     <Page
       id={id}
       header={<Header onBackBtnClick={handleClose} title="Group info" />}
     >
-      <RelativeContainer>
+      <RelativeContainer
+        onClick={() => {
+          setIsPopoverOpen(false);
```
Comment (bamnenim-kryptonium) [inline packages/beoble/components/Organisms/PanelPages/RightPanels/ChatroomInfo/index.tsx:339]: check useClickOutside

File: packages/beoble/components/Organisms/PanelPages/RightPanels/ChatroomInfoEdit/index.tsx [hunk 12]
```diff
@@ -0,0 +1,152 @@
...
+      .max(30, 'Display name cannot be longer than 30 characters.'),
+    chatroomDescription: yup
+      .string()
+      .optional()
+      .max(150, 'Group description cannot be more than 150 characters long.'),
+  })
+  .required();
+
+type ChatroomInfoUpdateFormType = yup.InferType<typeof schema>;
+export type chatroomEditPageType = 'DEFAULT' | 'SELECT_IMAGE';
```
Comment (bamnenim-kryptonium) [inline packages/beoble/components/Organisms/PanelPages/RightPanels/ChatroomInfoEdit/index.tsx:33]: the type should be defined in ProfileUpdate

File: packages/beoble/components/Organisms/PanelPages/RightPanels/ChatroomInfoEdit/index.tsx [hunk 13]
```diff
@@ -0,0 +1,152 @@
...
+        defaultValue: chatroom?.description,
+      },
+    ];
+  }, [chatroom]);
+
+  const handleOnSubmit = useCallback(
+    async (data: ChatroomInfoUpdateFormType, profileImgToUpdate) => {
+      if (chatroom && user) {
+        const putChatRoomBody: IPutChatRoomBody = {
+          ...(data.chatroomDisplayName !== chatroom?.display_name && {
```
Comment (bamnenim-kryptonium) [inline packages/beoble/components/Organisms/PanelPages/RightPanels/ChatroomInfoEdit/index.tsx:70]: just use data's value with isDirty formState in useForm

File: packages/beoble/components/Organisms/PanelPages/RightPanels/ChatroomInfoEdit/index.tsx [hunk 14]
```diff
@@ -0,0 +1,152 @@
...
+          chatroom: chatroom,
+          putChatRoomBody: putChatRoomBody,
+        });
+      }
+      back();
+    },
+    [Beoble.attachment, back, chatroom, putChatroom, user]
+  );
+
+  const handleChatroomAvatarPreview = useCallback(
```
Comment (bamnenim-kryptonium) [inline packages/beoble/components/Organisms/PanelPages/RightPanels/ChatroomInfoEdit/index.tsx:109]: rm

File: packages/beoble/components/Organisms/PanelPages/RightPanels/ChatroomInfoEdit/index.tsx [hunk 15]
```diff
@@ -0,0 +1,152 @@
...
+          title="Edit Group Info"
+        />
+      }
+    >
+      <ProfileUpdate
+        editPage={editPage}
+        setEditPage={setEditPage}
+        inputFormSchema={schema}
+        inputFormData={inputFormData}
+        avatarPreviewHandler={handleChatroomAvatarPreview}
```
Comment (bamnenim-kryptonium) [inline packages/beoble/components/Organisms/PanelPages/RightPanels/ChatroomInfoEdit/index.tsx:143]: pass default

**Review by bamnenim-kryptonium — CHANGES_REQUESTED** [review]

File: packages/beoble/components/OrganSystems/ProfileUpdate/index.tsx [hunk 16]
```diff
@@ -0,0 +1,196 @@
...
+  onSubmitBtnClickHandler,
+}: ProfileUpdateProps) => {
+  const { user } = useMainUser();
+
+  const handleImageEditPageBtnClick = () => {
+    setEditPage('SELECT_IMAGE');
+  };
+
+  const [profileImgToUpdate, setProfileImageToUpdate] = useState<
+    string | INFT | File
```
Comment (bamnenim-kryptonium) [inline packages/beoble/components/OrganSystems/ProfileUpdate/index.tsx:92]: UpdateProfileImgType

File: packages/beoble/components/OrganSystems/ProfileUpdate/index.tsx [hunk 17]
```diff
@@ -0,0 +1,196 @@
...
+  >();
+
+  const handleImageChange = (image: string | INFT) => {
```
Comment (bamnenim-kryptonium) [inline packages/beoble/components/OrganSystems/ProfileUpdate/index.tsx:95]: image can be File?

File: packages/beoble/components/OrganSystems/ProfileUpdate/index.tsx [hunk 18]
```diff
@@ -0,0 +1,196 @@
...
+      ...inputFormData.map((datum) => ({ [datum.name]: datum.defaultValue }))
+    ),
+  });
+
+  useEffect(() => {
+    //validate on mount with default value in user obj
+    trigger();
+  }, [user, trigger]);
+
+  const handleAvatarPreview = () => {
```
Comment (bamnenim-kryptonium) [inline packages/beoble/components/OrganSystems/ProfileUpdate/index.tsx:120]: useMemo

**Review by bamnenim-kryptonium — CHANGES_REQUESTED** [review]

**Review by bamnenim-kryptonium — APPROVED** [review]

**Pull request merged**